ocaml/sdk-gen/go/test_data/time_convert.go

```diff
@@ -1,4 +1,31 @@
-var timeFormats = []string{time.RFC3339, "20060102T15:04:05Z", "20060102T15:04:05"}
+var timeFormats = []string{
```

Comment:
Not caused by this PR, but why do we need to redefine the time formats in the test code?

Reply:
Not sure, I think it's there to ensure SDK source files are generated as expected.

@minglumlu could you help clarify?

Reply:
This `time_convert.go` is an expected string output of rendering the template `ConvertTime.mustache`. The output will be put into the `convert.go` in the SDK.

Reply:
So it's just a test of the mustache rendering, right?

Reply:
Yes, as it is under the `test_data` directory.

Reply:
Ah ok, they're test data in this case, not formats.
Now that I'm thinking about it, it might be nicer to use patterns as date formats in the mustache template instead of hardcoded dates, even if Go works with both. It looks to me like a better coding style - thoughts?

Reply:
actually now that I've had a look in Go we use hardcoded dates (see https://pkg.go.dev/time#Parse)

Below is a snippet for testing (with which I actually just spotted a typo!), you can run it in 
https://go.dev/play/ if you want


<details>

<summary>`example.go`</summary>

```go
package main

import (
	"fmt"
	"time"
)

func main() {
	dates := map[string]time.Time{
		// no dashes, no colons
		"20220101T123045":       time.Date(2022, 1, 1, 12, 30, 45, 0, time.Local),
		"20220101T123045Z":      time.Date(2022, 1, 1, 12, 30, 45, 0, time.UTC),
		"20220101T123045+03":    time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)), // +03 timezone
		"20220101T123045+0300":  time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),
		"20220101T123045+03:00": time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),

		"20220101T123045.123":       time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.Local),
		"20220101T123045.123Z":      time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.UTC),
		"20220101T123045.123+03":    time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.FixedZone("", 3*60*60)),
		"20220101T123045.123+0300":  time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.FixedZone("", 3*60*60)),
		"20220101T123045.123+03:00": time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.FixedZone("", 3*60*60)),

		// no dashes, with colons
		"20220101T12:30:45":       time.Date(2022, 1, 1, 12, 30, 45, 0, time.Local),
		"20220101T12:30:45Z":      time.Date(2022, 1, 1, 12, 30, 45, 0, time.UTC),
		"20220101T12:30:45+03":    time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),
		"20220101T12:30:45+0300":  time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),
		"20220101T12:30:45+03:00": time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),

		"20220101T12:30:45.123":       time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.Local),
		"20220101T12:30:45.123Z":      time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.UTC),
		"20220101T12:30:45.123+03":    time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.FixedZone("", 3*60*60)),
		"20220101T12:30:45.123+0300":  time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.FixedZone("", 3*60*60)),
		"20220101T12:30:45.123+03:00": time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.FixedZone("", 3*60*60)),

		// dashes and colons
		"2022-01-01T12:30:45":       time.Date(2022, 1, 1, 12, 30, 45, 0, time.Local),
		"2022-01-01T12:30:45Z":      time.Date(2022, 1, 1, 12, 30, 45, 0, time.UTC),
		"2022-01-01T12:30:45+03":    time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),
		"2022-01-01T12:30:45+0300":  time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),
		"2022-01-01T12:30:45+03:00": time.Date(2022, 1, 1, 12, 30, 45, 0, time.FixedZone("", 3*60*60)),

		"2022-01-01T12:30:45.123":    time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.Local),
		"2022-01-01T12:30:45.123Z":   time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.UTC),
		"2022-01-01T12:30:45.123+03": time.Date(2022, 1, 1, 12, 30, 45, 123000000, time.FixedZone("", 3*60*60)),
	}
	var timeFormats = []string{
		time.RFC3339,
		"2006-01-02T15:04:05",
		// no dashes, no colons
		"20060102T15:04:05Z",
		"20060102T15:04:05",
		"20060102T150405.999999999Z0700",
		"20060102T150405",
		"20060102T150405Z07",
		"20060102T150405Z07:00",
		// no dashes, with colons
		"20060102T15:04:05Z07",
		"20060102T15:04:05Z0700",
		"20060102T15:04:05Z07:00",
		"20060102T15:04:05.999999999Z07",
		"20060102T15:04:05.999999999Z07:00",
		"20060102T15:04:05.999999999Z07",
		// dashes and colon patterns not covered by `time.RFC3339`
		"2006-01-02T15:04:05Z07",
		"2006-01-02T15:04:05Z0700",
		"2006-01-02T15:04:05Z07:00",
		"2006-01-02T15:04:05.999999999Z07",
		"2006-01-02T15:04:05.999999999Z07:00",
		"2006-01-02T15:04:05.999999999Z07",
	}
	for input, expected := range dates {
		for _, timeFormat := range timeFormats {
			result, err := time.Parse(timeFormat, input)
			if err == nil {

				matching := expected.Equal(result)
				if !matching {
					fmt.Printf("Found for '%s'\n", input)
					fmt.Printf("Corresponding time format is: '%s'\n", timeFormat)
					fmt.Printf("Expected: %s\n", expected)
					fmt.Printf("Parsed: %s\n\n", result)
				}
			}

		}
	}

}
```
</details>